FixMath.h

```diff
@@ -0,0 +1,1287 @@
+
+template <int8_t NI, int8_t NF>
+inline SFixMath<NI, NF> operator-(uint64_t op, const UFixMath<NI, NF>& uf) {return -uf+op;}
+
+template <int8_t NI, int8_t NF>
+inline SFixMath<NI, NF> operator-(int8_t op, const UFixMath<NI, NF>& uf) {return -uf+op;}
+
+template <int8_t NI, int8_t NF>
+inline SFixMath<NI, NF> operator-(int16_t op, const UFixMath<NI, NF>& uf) {return -uf+op;}
+
+template <int8_t NI, int8_t NF>
+inline SFixMath<NI, NF> operator-(int32_t op, const UFixMath<NI, NF>& uf) {return -uf+op;}
+
+template <int8_t NI, int8_t NF>
+inline SFixMath<NI, NF> operator-(int64_t op, const UFixMath<NI, NF>& uf) {return -uf+op;}
+
+template <int8_t NI, int8_t NF>
+inline SFixMath<NI, NF> operator-(float op, const UFixMath<NI, NF>& uf) {return -uf+op;}
+
+template <int8_t NI, int8_t NF>
+inline SFixMath<NI, NF> operator-(double op, const UFixMath<NI, NF>& uf) {return -uf+op;}
+#endif
+////// Helper functions to build SFix from a normal type automatically
+
+
+/** Create a *pure* fractional unsigned fixed number (UFixMath) from an unsigned integer.
+    The number of fractional bits (NF) is chosen automatically depending on the input 
+    type. Hence toUFraction(255) and toUFraction(uint8_t(255)) *do not* lead to the 
+    same thing: on an AVR, the former will lead to NF=16 - which is overkill and 
+    incorrect if you expect toUFraction(255) = 1 -
+    whereas the latter will lead to NF=8. Mozzi's objects (Oscil and the like)
+    returns correct types, hence you can use this function to convert the return
+    value of a Mozzi's function/class member into a pure fractional number.
+    @param val The value to be converted into a pure fractional number.
+    @return A UFixMath<0,NF> with NF chosen according to the input type
+*/
+template<typename T>
+inline UFixMath<0, sizeof(T)*8> toUFraction(T val) {
+  return UFixMath<0, sizeof(T)*8>::fromRaw(val); 
+}
+
+/** Create a *pure* integer unsigned fixed number (UFixMath) from an unsigned integer.
+    The number of fractional bits (NI) is chosen automatically depending on the input 
+    type. Hence toUInt(255) and toSInt(uint8_t(255)) *do not* lead to the 
+    same thing: on an AVR, the former will lead to NI=16 - which is overkill -
+    whereas the latter will lead to NI=8. Mozzi's objects (Oscil and the like)
+    returns correct types, hence you can use this function to convert the return
+    value of a Mozzi's function/class member into a pure fractional number.
+    @param val The value to be converted into a pure unsigned integer fixed math number.
+    @return A UFixMath<NI,0> with NI chosen according to the input type
+*/
+template<typename T>
+inline UFixMath<sizeof(T)*8-1,0> toUInt(T val) {
+  return UFixMath<sizeof(T)*8-1,0>::fromRaw(val); 
+}
+
+
+
+/** Instanciate an signed fixed point math number.
+    @param NI The number of bits encoding the integer part. The integral part can range into [-2^NI, 2^NI -1]
+    @param NF The number of bits encoding the fractional part
```

Comment:
(Also for UFixMath): AFAIU, RANGE is still "visible" (I for one don't know a way to hide it). Maybe it makes sense to document it as "purely internal"?